sled-agent/src/services.rs

```diff
@@ -1074,6 +1085,40 @@ impl ServiceManager {
                 let Some(info) = self.inner.sled_info.get() else {
                     return Err(Error::SledAgentNotReady);
                 };
+
+                // We want to configure the dns/install SMF service inside the
```

Comment:
Should we pull this out into a separate method? I'd be shocked if this was the only service for which we wanted to populate `/etc/resolv.conf`.

(I mean, arguably, shouldn't we populate basically all our services with `/etc/resolv.conf`?)

Reply:
I do think we should do this, but it's more complicated to use it:
https://github.com/oxidecomputer/omicron/issues/2122#issuecomment-1611870486

so I'd like to defer it for now.